nntrainer/src/bn_layer.cpp

```diff
@@ -34,13 +34,23 @@ namespace nntrainer {
 
 enum class BNParams { mu, var, gamma, beta };
 
-/// @todo add channel wise bn for convolutional layer.
+/// @todo add multiple axis support
 int BatchNormalizationLayer::initialize() {
   int status = ML_ERROR_NONE;
 
   output_dim = input_dim;
-  TensorDim dim = input_dim;
-  dim.batch(1);
+  TensorDim dim;
+
+  /// @note this logic cannot tell channel is actually 1 or it is just not used.
+  if (axis == -1)
+    axis = input_dim.channel() > 1 ? 1 : 3;
+
+  dim.setTensorDim(axis, input_dim.getTensorDim(axis));
```

Comment:
with batch?

Reply:
The batch should be 1 for params isn't it?

Reply:
aah input_dim is now without batch.